simplesource-data/src/test/java/io/simplesource/data/FutureResultTest.java

```diff
@@ -15,10 +15,15 @@
 class FutureResultTest {
     private static final int COUNTDOWN_SLEEP_TIME = 1000;
     private static final String DEFAULT_VALUE = "not exist value";
-    private static final Reason<CommandError> FAILURE_REASON_1 = Reason.of(CommandError.InternalError, "Internal error");
-    private static final Reason<CommandError> FAILURE_REASON_2 = Reason.of(CommandError.InvalidCommand, "Invalid command error");
-    private static final Function<NonEmptyList<Reason<CommandError>>, List<CommandError>> CONCATENATE_ERROR_REASONS_FUNC =
-            rs -> rs.stream().map(Reason::getError).collect(Collectors.toList());
+    private static final io.simplesource.data.CommandError FAILURE_COMMAND_ERROR_1 =
```

Comment:
We can run the test with any error data type. It can be `String`.

Reply:
I wanted to be as close as possible to the actual use of `FutureResult.fold` where client will need to map `CommandError` to `BusinessError`. Happy to change it to a new type `TestError`